src/content/blog/astro-290.mdx

```diff
@@ -0,0 +1,141 @@
+
+export default defineConfig({
+  experimental: {
+    viewTransitions: true,
+  },
+});
+```
+
+This enables you to use the new APIs added.
+
+### &lt;ViewTransitions /&gt;
+
+This is a component which acts as the _router_ for transitions between pages. Add it to the `<head>` section of each individual page where transitions should occur _in the client_ as you navigate away to another page, instead of causing a full page browser refresh. To enable support throughout your entire app, add the component in some common layout or component that targets the `<head>` of every page.
+
+**CommonHead.astro**
+
+```astro
+---
+import { ViewTransitions } from 'astro:transitions';
+---
+
+<meta charset="utf-8" />
+<title>{Astro.props.title}</title>
+<ViewTransitions />
+```
+
+With only this change, your app will now route completely in-client. You can then add transitions to individual elements using the `transition:animate` directive.
+
+### Animations
+
+Add `transition:animate` to any element to use Astro's built-in animations.
+
+```astro
+<header transition:animate="slide"></header>
+```
+
+In the above, Astro's `slide` animation will cause the `<header>` element to slide out to the left, and then slide in from the right when you navigate away from the page.
+
+You can also customize these animations using any CSS animation properties, for example, by specifying a duration:
+
+```astro
+---
+import { slide } from 'astro:transition';
+---
+
+<header transition:animate={slide({ duration: 200 })}></header>
+```
+
+### Continue learning
+
+Check out the [client-side routing docs](https://docs.astro.build/en/guides/client-side-routing/) to learn more.
+
+## Redirects
+
+This change moves the `redirects` configuration out of experimental. If you were previously using experimental redirects, remove the following experimental flag:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    redirects: true,
+  }
+});
+```
+
+If you have been waiting for stabilization before using redirects, now you can do so. Check out [the docs on redirects](https://docs.astro.build/en/core-concepts/routing/#redirects) to learn how to use this built-in feature.
+
+A basic example is this, in your configuration, add a map of paths to redirect from and to:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  redirects: {
+    '/old-page': '/new-page'
+  }
+});
+```
+
+## Hoisted script optimization
+
+Astro's static analysis to determine which `<script>` tags to bundle together just got a little smarter!
+
+Astro create bundles that optimize script usage between pages and place them in the head of the document so that they are downloaded as early as possible. One limitation to Astro's existing approach has been that you could not dynamically use hoisted scripts. Each page received the same, all-inclusive bundle whether or not every script was needed on that page.
```

Comment:
```suggestion
Astro create bundles that optimize script usage between pages and place them in the head of the document so that they are downloaded as early as possible. One limitation to Astro's existing approach has been that you could not dynamically use hoisted scripts. When using libraries that re-exported components, each page received the same, all-inclusive bundle whether or not every script was needed on that page.
```
I *think* this is what the PR author was getting at in the original PR when they added a comment after merging! Double-check for nuance!

Reply:
There's also a verbal disagreement in this line, adding this on top of Sarah's suggestion!
```suggestion
Astro creates bundles that optimize script usage between pages and place them in the head of the document so that they are downloaded as early as possible. One limitation to Astro's existing approach has been that you could not dynamically use hoisted scripts. When using libraries that re-exported components, each page received the same, all-inclusive bundle whether or not every script was needed on that page.
```